tests/load/clickhouse/test_clickhouse_adapter.py

```diff
@@ -25,37 +36,50 @@ def not_annotated_resource():
     assert_load_info(pack)
 
     with pipe.sql_client() as client:
-        # get map of table names to full table names
+        # Get a map of table names to full table names.
         tables = {}
         for table in client._list_tables():
             if "resource" in table:
                 tables[table.split("___")[1]] = table
         assert (len(tables.keys())) == 3
 
-        # check content
+        # Check the table content.
         for full_table_name in tables.values():
             with client.execute_query(f"SELECT * FROM {full_table_name};") as cursor:
                 res = cursor.fetchall()
                 assert tuple(res[0])[:2] == (1, 2)
 
-        # check table format
-        # fails now, because we do not have a cluster (I think), it will fall back to SharedMergeTree
-        for full_table_name in tables.values():
+        # Check the table engine.
+        for table_name, full_table_name in tables.items():
             with client.execute_query(
-                "SELECT database, name, engine, engine_full FROM system.tables WHERE name ="
-                f" '{full_table_name}';"
+                "SELECT database, name, engine, engine_full FROM system.tables "
+                f"WHERE name = '{full_table_name}';"
             ) as cursor:
                 res = cursor.fetchall()
-                # this should test that two tables should be replicatedmergetree tables
-                assert tuple(res[0])[2] == "SharedMergeTree"
+                if table_name in (
+                    "merge_tree_resource",
+                    "replicated_merge_tree_resource",
+                ):
+                    assert tuple(res[0])[2] in (
+                        "MergeTree",
+                        "SharedMergeTree",
+                        "ReplicatedMergeTree",
+                    )
+                else:
+                    # Non annotated resource needs to default to detected installation
+                    # type, i.e. cloud or self-managed.
+                    # CI runs on CH cloud, so will be `SharedMergeTree`.
```

Comment:
Should we run CI both for ClickHouse cloud and ClickHouse local, like Postgres?

Seems like we don't have proper test coverage without local (e.g. for the `Log` table engines).

Reply:
- Added ClickHouse OSS as a local destination test in `.github/workflows/test_local_destinations.yml`, so CI also runs local tests.
- Removed the log engine family for now. I'm not even sure why I brought it into this ticket, but it will require more work than what can be done here.